copy/copy.go

```diff
@@ -813,6 +814,17 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 		return types.BlobInfo{}, errors.Wrap(err, "Error writing blob")
 	}
 
+	switch compressionOperation {
+	case types.Compress:
+		uploadedInfo.MediaType = manifest.DockerV2Schema2LayerMediaType
+	case types.Decompress:
+		uploadedInfo.MediaType = manifest.DockerV2SchemeLayerMediaTypeUncompressed
```

Comment:
This needs to be specific to the schema (`manifest.Manifest`/`types.Image`) implementation: OCI uses differently-named types.

(And it probably only should change MIME types if the previous value is exactly what is expected, so that a config MIME type is not replaced by a blob MIME type.)

Reply:
> This needs to be specific to the schema (`manifest.Manifest`/`types.Image`) implementation: OCI uses differently-named types.

:+1: 

> (And it probably only should change MIME types if the previous value is exactly what is expected, so that a config MIME type is not replaced by a blob MIME type.)

I don't think that's possible, as the code in question only applies to blobs.

Reply:
`copyConfig` calls `copyBlobFromStream` — though you are right that it does not matter, because it sets `canModifyBlob` to `false`.